rs/ic_os/config/src/lib.rs

```diff
@@ -113,27 +115,40 @@ mod tests {
             guestos_settings: guestos_settings.clone(),
         };
 
-        let temp_dir = tempdir().expect("Failed to create temp directory");
-        let json_setupos_file = temp_dir.path().join("test_setupos_config.json");
-        let json_hostos_file = temp_dir.path().join("test_hostos_config.json");
-        let json_guestos_file = temp_dir.path().join("test_guestos_config.json");
-
-        assert!(serialize_and_write_config(&json_setupos_file, &setupos_config_struct).is_ok());
-        assert!(serialize_and_write_config(&json_hostos_file, &hostos_config_struct).is_ok());
-        assert!(serialize_and_write_config(&json_guestos_file, &guestos_config_struct).is_ok());
+        // Test serialization of SetupOSConfig
+        let mut buffer = Vec::new();
+        serde_json::to_writer_pretty(&mut buffer, &setupos_config_struct)
+            .expect("Failed to serialize SetupOS config");
+        assert!(!buffer.is_empty());
 
+        // Test deserialization of SetupOSConfig
+        let mut cursor = Cursor::new(buffer);
```

Comment:
Why do you need a cursor? &[u8] implements Read (or you can also use from_slice).

Reply:
Ah, much nicer! I created a helper function while I was at it. See: https://github.com/dfinity/ic/pull/1539